RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClusterizerConditionsESProducerAlpaka.cc

```diff
@@ -0,0 +1,227 @@
+      const int Data_apv_size = gain_.size();
+      //// Typical sizes for these collections (from MC run)
+      //// constexpr const unsigned int DetToFeds_size= 34813;
+      //// constexpr const unsigned int Data_fedch_size= 42240;
+      //// constexpr const unsigned int Data_strip_size= 10813440;
+      //// constexpr const unsigned int Data_apv_size= 84480;
+
+      assert(detID_.size() == iPair_.size() && iPair_.size() == invthick_.size());
+
+      // PortableHostMultiCollection(const std::array<int32_t, members_>& sizes, alpaka_common::DevHost const& host)
+      auto product = std::make_unique<SiStripClusterizerConditionsHost>(
+          std::array<int32_t, 4>{{DetToFeds_size, Data_fedch_size, Data_strip_size, Data_apv_size}},
+          cms::alpakatools::host());
+
+      auto DetToFeds_View = product->view();
+      auto Data_fedch_View = product->view<SiStripClusterizerConditionsData_fedchSoA>();
+      auto Data_strip_View = product->view<SiStripClusterizerConditionsData_stripSoA>();
+      auto Data_apv_View = product->view<SiStripClusterizerConditionsData_apvSoA>();
+
+      for (int j = 0; j < DetToFeds_size; ++j) {
+        auto entry = detToFeds_[j];
+        DetToFeds_View[j].detid_() = entry.detID();
+        DetToFeds_View[j].fedch_() = entry.fedCh();
+        DetToFeds_View[j].fedid_() = entry.fedID();
+        DetToFeds_View[j].ipair_() = entry.pair();
+      }
+      for (int j = 0; j < Data_fedch_size; ++j) {
+        Data_fedch_View[j].detID_() = detID_[j];
+        Data_fedch_View[j].iPair_() = iPair_[j];
+        Data_fedch_View[j].invthick_() = invthick_[j];
+      }
+      for (int j = 0; j < Data_strip_size; ++j) {
+        Data_strip_View[j].noise_() = noise_[j];
+      }
+      for (int j = 0; j < Data_apv_size; ++j) {
+        Data_apv_View[j].gain_() = gain_[j];
+      }
+
+      return product;
+    }
+
+    // Auxiliary functions to translate indexes on the arrays
+    static constexpr uint16_t badBit = 1 << 15;
+    inline uint16_t fedIndex(uint16_t fed) { return (fed - FED_ID_MIN); };
+    inline uint32_t stripIndex(uint16_t fed, uint8_t channel, uint16_t strip) {
+      return (fedIndex(fed) * FEDCH_PER_FED * STRIPS_PER_FEDCH + channel * STRIPS_PER_FEDCH +
+              (strip % STRIPS_PER_FEDCH));
+    };
+    inline uint32_t apvIndex(uint16_t fed, uint8_t channel, uint16_t strip) {
+      return (fedIndex(fed) * APVS_PER_FEDCH * FEDCH_PER_FED + APVS_PER_CHAN * channel +
+              (strip % STRIPS_PER_FEDCH) / STRIPS_PER_APV);
+    };
+    inline uint32_t channelIndex(uint16_t fed, uint8_t channel) { return (fedIndex(fed) * FEDCH_PER_FED + channel); };
+    inline void setInvThickness(uint16_t fed, uint8_t channel, float invthick) {
+      invthick_[channelIndex(fed, channel)] = invthick;
+    };
+
+  private:
+    std::vector<float> invthick_;
+    std::vector<uint32_t> detID_;
+    std::vector<uint16_t> iPair_;
+    std::vector<uint16_t> noise_;
+    std::vector<float> gain_;
```

Comment:
It would be better to store this temporary data in local variables in the `produce()` function instead of member variables. That would make the flow of data more clear (e.g. the purpose of `SiStripClusterizerConditionsOnHost_` would be more clear), and avoid storing these data when they are not needed (i.e. outside of the produce()` function calls).

Or maybe the functions could fill the `SiStripClusterizerConditionsHost` directly without intermediate vectors?